xonsh/main.py

```diff
@@ -224,8 +224,8 @@ def main(argv=None):
         # run a script contained in a file
         path = os.path.abspath(os.path.expanduser(args.file))
         if os.path.isfile(path):
-            sys.argv = args.args
-            env['ARGS'] = [args.file] + args.args
+            sys.argv = [args.file] + args.args
+            env['ARGS'] = sys.argv[:]  # $ARGS is not sys.argv
```

Comment:
There is no clear relationship between `$ARGS` environmental variable and `sys.argv` global list (except that both hold the same elements at the startup). I doubt it's evident that any changes made to one of them should apply to the other as well. Explicit is better than implicit.